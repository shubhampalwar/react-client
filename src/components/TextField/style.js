const style = {
  base: {
    height: '20px',
    padding: '10px',
    width: '100%',
    borderRadius: '5px',
    border: '1px solid lightgrey',
  },
  error: {
    border: '1px solid red',
  },
};

export default style;
