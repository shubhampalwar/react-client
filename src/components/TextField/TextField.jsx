import React, { Component } from 'react';
import PropTypes from 'prop-types';
import style from './style';

class TextField extends Component {
  constructor(props) {
    super(props);
    this.props = props;
  }

  render() {
    const { error, title, ...rest } = this.props;
    const errorStyle = error ? style.error : {};
    return (
      <>
        <div style={style.title}>{title}</div>
        <input type="text" {...rest} style={{ ...style.base, ...errorStyle }} />
        <div style={style.errorText}>{error}</div>
      </>
    );
  }
}

TextField.propTypes = {
  error: PropTypes.string,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
};
TextField.defaultProps = {
  error: '',
};
export default TextField;
