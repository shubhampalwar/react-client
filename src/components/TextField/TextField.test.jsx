import { shallow } from 'enzyme';
import React from 'react';
import TextField from './TextField';

const wrapper = shallow(<TextField />);

describe('TextField', () => {
  it('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
