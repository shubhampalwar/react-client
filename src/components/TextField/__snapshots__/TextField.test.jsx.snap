// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextField
    error=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          style={
            Object {
              "fontSize": "21px",
              "fontWeight": "bold",
            }
          }
        />,
        <input
          id="textField"
          style={
            Object {
              "border": "1px solid lightgrey",
              "borderRadius": "5px",
              "boxSizing": "border-box",
              "margin": "5px 0px 0px 0px",
              "padding": "12px",
              "width": "100%",
            }
          }
          type="text"
        />,
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "color": "red",
              "padding": "10px",
            }
          }
        >
          
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "style": Object {
            "fontSize": "21px",
            "fontWeight": "bold",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "textField",
          "style": Object {
            "border": "1px solid lightgrey",
            "borderRadius": "5px",
            "boxSizing": "border-box",
            "margin": "5px 0px 0px 0px",
            "padding": "12px",
            "width": "100%",
          },
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "boxSizing": "border-box",
            "color": "red",
            "padding": "10px",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            style={
              Object {
                "fontSize": "21px",
                "fontWeight": "bold",
              }
            }
          />,
          <input
            id="textField"
            style={
              Object {
                "border": "1px solid lightgrey",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "margin": "5px 0px 0px 0px",
                "padding": "12px",
                "width": "100%",
              }
            }
            type="text"
          />,
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "color": "red",
                "padding": "10px",
              }
            }
          >
            
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "style": Object {
              "fontSize": "21px",
              "fontWeight": "bold",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "textField",
            "style": Object {
              "border": "1px solid lightgrey",
              "borderRadius": "5px",
              "boxSizing": "border-box",
              "margin": "5px 0px 0px 0px",
              "padding": "12px",
              "width": "100%",
            },
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "boxSizing": "border-box",
              "color": "red",
              "padding": "10px",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
